src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -36,7 +36,7 @@ public void Add(Feature feature)
             }));
         }
 
-        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings)
+        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings, RoutingComponent routing)
```

Comment:
@SzymonPobiega I actually like it. I think we went a bit too far with componentization in the core. Some things just have to be in the core in order to deliver a meaningful product which is self contained. I think routing is an integral piece of the core like the transport which deserves to be treated special like that.